src/AccessLogs.sol

```diff
@@ -0,0 +1,190 @@
+// Copyright 2023 Cartesi Pte. Ltd.
+
+// SPDX-License-Identifier: Apache-2.0
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+/// @dev This file is generated from helper_scripts/generate_AccessLogs.sh, one should not modify the content directly
+
+pragma solidity ^0.8.0;
+
+import "./Memory.sol";
+import "./UArchConstants.sol";
+
+library AccessLogs {
+    using AccessLogs for bytes;
+    using Memory for Memory.AlignedSize;
+
+    struct Context {
+        bytes32 currentRootHash;
+        bytes buffer;
+        uint128 pointer;
+    }
+
+    /// @notice Swap byte order of unsigned ints with 64 bytes
+    /// @param num number to have bytes swapped
+    function uint64SwapEndian(uint64 num) internal pure returns (uint64) {
+        uint64 output = ((num & 0x00000000000000ff) << 56)
+            | ((num & 0x000000000000ff00) << 40)
+            | ((num & 0x0000000000ff0000) << 24) | ((num & 0x00000000ff000000) << 8)
+            | ((num & 0x000000ff00000000) >> 8) | ((num & 0x0000ff0000000000) >> 24)
+            | ((num & 0x00ff000000000000) >> 40)
+            | ((num & 0xff00000000000000) >> 56);
+
+        return output;
+    }
+
+    function writeBytes32(bytes memory data, uint128 offset, bytes32 val)
+        internal
+        pure
+    {
+        assembly {
+            mstore(add(data, add(offset, 32)), val)
+        }
+    }
+
+    function toBytes32(bytes memory data, uint128 offset)
```

Comment:
If we go for a buffer type, this function would receive a buffer. We could call it like this:

```
a.buffer.toBytes32()
```

Also, I think we can name this function better. What do you think of `readBytes32` and `peekBytes32`?